gtest-parallel

```diff
@@ -82,6 +87,45 @@ class RawFormat:
   def end(self):
     pass
 
+# Record of test runtimes. Has built-in locking.
+class TestTimes(object):
+  def __init__(self, save_file):
+    "Create new object seeded with saved test times from the given file."
+    self.__old_times = {}  # (test binary, test name) -> runtime in seconds
```

Comment:
Do you need both old_times and new_times? get_test_time is only called before tests are running, right?

Also, imo remove the __ prefixes, unless you can point me to somewhere saying they're Google style.